### ISSUE TITLE
inconstistent behavior with colors for `geom_coverage`

### ISSUE THREAD

**Issue by m-jahn** [showteeth/ggcoverage#15]
Hi,

I noticed some problem with the `color` flag in `geom_coverage`.
The documentation is a little sparse on what arguments exactly it takes, so I needed to try a lot to find out. I used the example from the docs page, and this is what I noticed.

Loading  example data:

```
library(ggcoverage)
meta.file <- system.file("extdata", "RNA-seq", "meta_info.csv", package = "ggcoverage")
sample.meta <- utils::read.csv(meta.file)

track.folder <- system.file("extdata", "RNA-seq", package = "ggcoverage")

track.df <- LoadTrackFile(
  track.folder = track.folder, format = "bw",
  meta.info = sample.meta
)
```

1. example `color = "auto"` has no effect, like any other string. This produces a warning.

```
ggplot() +
  geom_coverage(data = track.df, color = "auto")
```

2. supplying two colors for two groups does not have desired effect. I would think that you could subdivide the data by facets/tracks, and use same or different colors for those, but that doesn't work. Falls back to some default scheme

```
ggplot() +
  geom_coverage(data = track.df, color = c("#E7298A", "#66A61E"))
```

3.  supplying more colors than needed also does not have desired effect, again another palette is used

```
ggplot() +
  geom_coverage(data = track.df, color = c("#E7298A", "#66A61E", "#E6AB02"))
```

Suggestion for fixes:

- remove "auto" from examples, replace with NULL
- check that supplied colors are in fact valid
- make number of colors match number of groups ("Group" variable), not tracks ("Type" variable)
- allow supplying more colors than needed for groups, but not less --> raise warning

I implemented those fixes in a commit on my fork. I can make a PR if you like. You can test it here:
https://github.com/m-jahn/ggcoverage

- this will work as expected now, with  tracks and groups being handled separately

```
ggplot() +
  geom_coverage(data = track.df, color = c("#E7298A", "#66A61E"))
```

![image](https://github.com/showteeth/ggcoverage/assets/32667667/604ad63a-57f0-4c3e-8787-271d128298b7)

- if you want different colors for each track, just use the same variable for tracks and groups

```
ggplot() +
  geom_coverage(
    data = track.df,
    group.key = "Type", facet.key = "Type",
    color = c("#E7298A", "#66A61E", "#E6AB02", "#7570B3")
  )
```

![image](https://github.com/showteeth/ggcoverage/assets/32667667/f5753496-b283-47f5-8116-4f045b536ffb)

- supplying too few colors raises a warning and falls back to a default scale

```
ggplot() +
  geom_coverage(
    data = track.df,
    group.key = "Type",
    color = c("#E7298A", "#66A61E", "#E6AB02")
  )
```

- supplying more colors is not a problem

```
ggplot() +
  geom_coverage(
    data = track.df,
    color = heat.colors(4)
  )
```

- supplying invalid colors raises an error

```
ggplot() +
  geom_coverage(data = track.df, color = c("red", "notacolor"))
```

**Comment by showteeth** [comment]
@m-jahn 

Thanks for your suggestions, they are awesome!

In the initial design, for different tracks, users should ensure that the number of colors provided and tracks is the same, otherwise `ggcoverage` will raise warning and automatically select colors. In short, `facet.key` (tracks)-based.

I took a look at your modifications, it seems that you changed the `facet.key` (tracks)-based to `group.key` (group)-based mode. Thus, users can not color every track color, it this true? Ideally, users can choose  `facet.key` (tracks)-based or `group.key` (group)-based mode (add an `ifelse` to check the color numbers, for numbers smaller than `facet.key` numbers, try group-based mode, otherwise track-based mode). What do you think about this?

Please feel free to create PR!

YB

**Comment by m-jahn** [comment]
I see your point! I think ideally the user should have the possibility to be as flexible as possible.
The idea from base ggplot2 is that one uses facets with one variable, and colors (or symbols) for other variables. That's why I attempted to change the coloring to the `group.key` instead of `facet.key` variable, as you pointed out. Users can still color every track differently by using the same variable for tracks/facets and for grouping (second plot from my original post). But you're right, it's not the default.

I can change it such that when no group variable is given, the different tracks are colored. Or as an alternative, without `group.key` specified the default color is used for all tracks, which would be more `ggplot`-like. I think I like that second option more, but it's up to you.

I just checked and it actually works out of the box to get the default color from theme when `group.key = NULL`.

With the example data, compare a plot _with_ grouping:

```
ggplot() +
  geom_coverage(data = track.df, facet.key = "Type", group.key = "Group")
```

![image](https://github.com/showteeth/ggcoverage/assets/32667667/777661d7-938a-4d60-b2c9-adb4e4938ec2)

to no grouping:

```
ggplot() +
  geom_coverage(data = track.df, facet.key = "Type", group.key = NULL)
```

![image](https://github.com/showteeth/ggcoverage/assets/32667667/dc536547-1398-439a-b492-9a49862ddf00)

and to coloring all facets by using same variable:

```
ggplot() +
  geom_coverage(data = track.df, facet.key = "Type", group.key = "Type")
``` 

![image](https://github.com/showteeth/ggcoverage/assets/32667667/cd18683a-93a5-4da8-9b3c-ee0c1e024c53)

This is how I would expect it to work.
